pkg/destroy/gcp/cloudcontroller.go

```diff
@@ -0,0 +1,201 @@
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_naming.go#L33-L40
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_clusterid.go#L210-L238
+func (o *ClusterUninstaller) listCloudControllerInstanceGroups() ([]cloudResource, error) {
+	filter := fmt.Sprintf("name eq \"k8s-ig--%s\"", o.cloudControllerUID)
+	return o.listInstanceGroupsWithFilter("items/*/instanceGroups(name,zone),nextPageToken", filter, nil)
+}
+
+// listBackendServicesForInstanceGroups returns backend services created by the cloud controller.
+// It list all backend services matching the cloud controller name convention that contain
+// only cluster instance groups.
+func (o *ClusterUninstaller) listBackendServicesForInstanceGroups(instanceGroups []cloudResource, installerInstanceGroups []cloudResource) ([]cloudResource, error) {
+	urls := sets.NewString()
+	// Allow instance groups created by the cloud controller
+	for _, instanceGroup := range instanceGroups {
+		urls.Insert(o.getInstanceGroupURL(instanceGroup))
+	}
+	// Also allow instance groups created by installer
+	for _, instanceGroup := range installerInstanceGroups {
+		urls.Insert(o.getInstanceGroupURL(instanceGroup))
+	}
+	filter := "name eq \"a[0-9a-f]{30,50}\""
+	return o.listBackendServicesWithFilter("items(name,backends),nextPageToken", filter, func(item *compute.BackendService) bool {
+		if len(item.Backends) == 0 {
+			return false
+		}
+		for _, backend := range item.Backends {
+			// Ignore this backend because it contains an unknown instance group.
+			if !urls.Has(backend.Group) {
+				return false
+			}
+		}
+		return true
+	})
+}
+
+// listCloudControllerTargetPools returns target pools created by the cloud controller.
+// It list all target pools matching the cloud controller name convention that contain
+// only cluster instances.
+func (o *ClusterUninstaller) listCloudControllerTargetPools() ([]cloudResource, error) {
+	filter := "name eq \"a[0-9a-f]{30,50}\""
+	return o.listTargetPoolsWithFilter("items(name,instances),nextPageToken", filter, func(pool *compute.TargetPool) bool {
+		if len(pool.Instances) == 0 {
+			return false
+		}
+		for _, instanceURL := range pool.Instances {
+			name, _ := o.getInstanceNameAndZone(instanceURL)
+			if !o.isClusterResource(name) {
+				return false
+			}
+		}
+		return true
+	})
+}
+
+// DiscoverLoadBalancerResources follows a similar procedure as:
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go#L222
+// https://github.com/openshift/kubernetes/blob/1e5983903742f64bca36a464582178c940353e9a/pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go#L289
+func (o *ClusterUninstaller) discoverLoadBalancerResources(loadBalancerName string) error {
```

Comment:
nit: this function name is too generic, `discoverCloudControllerresources` or something..

Reply:
Done.